Lib/typing.py

```diff
@@ -2338,12 +2335,12 @@ def get_type_hints(obj, globalns=None, localns=None, include_extras=False,
                 # This only affects ForwardRefs.
                 base_globals, base_locals = base_locals, base_globals
             for name, value in ann.items():
-                if value is None:
-                    value = type(None)
                 if isinstance(value, str):
                     value = _make_forward_ref(value, is_argument=False, is_class=True)
                 value = _eval_type(value, base_globals, base_locals, base.__type_params__,
                                    format=format, owner=obj)
+                if value is None:
+                    value = type(None)
```

Comment:
I thought in https://github.com/python/cpython/issues/133960 you said you wanted to remove the `None` -> `NoneType` conversion? Or am I misunderstanding what this does here?

Reply:
This is in typing.get_type_hints, for which I don't intend to change the publicly visible behavior. A consequence of some other part of this change is that `_eval_type` can now sometimes return None instead of NoneType, and we want get_type_hints() to still return NoneType.

The `test_pep_695_generic_method_with_future_annotations` test fails if I undo the changes in `get_type_hints`.